core/src/mindustry/ui/dialogs/PlanetDialog.java

```diff
@@ -405,19 +405,22 @@ void setup(){
 
         stack(
         new Element(){
+            long tapMillis;
+
             {
                 //add listener to the background rect, so it doesn't get unnecessary touch input
                 addListener(new ElementGestureListener(){
                     @Override
                     public void tap(InputEvent event, float x, float y, int count, KeyCode button){
                         if(showing()) return;
 
-                        if(selected == hovered){
+                        if(selected == hovered && Time.timeSinceMillis(tapMillis) < 500){
```

Comment:
`int count` already stored the amount of taps done, you shouldn't need to do your own calculations.